Source/cielim/Private/RenderingFunctionsLibrary.cpp

```diff
@@ -1,28 +1,58 @@
-
 #include "RenderingFunctionsLibrary.h"
 
 #include <OpenCV/PreOpenCVHeaders.h>
 #include <opencv2/core.hpp>
+#include "opencv2/imgproc.hpp"
 #include <OpenCV/opencv-4.5.5/modules/imgcodecs/include/opencv2/imgcodecs.hpp>
 #include <OpenCV/PostOpenCVHeaders.h>
 
+#include <filesystem>
+#include "CielimLoggingMacros.h"
+
+
 URenderingFunctionsLibrary::URenderingFunctionsLibrary(const FObjectInitializer& ObjectInitializer)
 {
-
+	
 }
 
-void URenderingFunctionsLibrary::ApplyPSF_Gaussian()
-
+FString URenderingFunctionsLibrary::ApplyPSF_Gaussian(FString Filepath, int32 KernelHeight, int32 KernelWidth, double SigmaX, double SigmaY)
 {
+	//NOTE: both dimensions of KernelSize must be odd
+	//Read Image
+
+	FString Filepath_Absolute = IFileManager::Get().ConvertToAbsolutePathForExternalAppForRead(*Filepath);
+	
+	std::string Filepath_Absolute_String = TCHAR_TO_UTF8(*Filepath_Absolute);
+	cv::Mat Image = cv::imread(Filepath_Absolute_String);
+
+	//Init ResultImage
```

Comment:
The attention to adding comments is great. However, let's be careful to only add comments when it aids understanding.

Reply:
Yes! I'll remove some of the more redundant comments